pkg/requests/requests.go

```diff
@@ -0,0 +1,78 @@
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// ConstructGetURL creates a GET request from a base URL, a map of query
+// parameters, and a formatted path.
+func ConstructGetURL(
+	rawURL string, queries map[string][]string, format string, values ...interface{}) (*url.URL, error) {
+
+	urlPath, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	urlPath.Path = fmt.Sprintf(format, values...)
+	var q url.Values = queries
+	urlPath.RawQuery = q.Encode()
+
+	return urlPath, nil
+}
+
+// GetAndParseJSON invokes a HTTP request with basic auth.
+// On success the response body will be unmarshalled as JSON.
+func GetAndParseJSON(result interface{}, user, token string, urlPath *url.URL) error {
```

Comment:
no need to export this function.

Reply:
Used outside of the requests package still.

Reply:
Ooops, sorry